core/errors/function-call-match-error.ts

```diff
@@ -1,8 +1,12 @@
 import { MatchError } from "../_errors";
+import { FunctionSpy } from "../_spying";
 
 export class FunctionCallMatchError extends MatchError {
 
-  public constructor(actualValue: any, shouldMatch: boolean, args?: Array<any>) {
-    super(actualValue, "function to be called", `Expected function ${!shouldMatch ? "not " : ""}to be called${args ? " with [" + args.map(arg => JSON.stringify(arg)).join(", ") + "]" : ""}.`);
+  public constructor(actualValue: FunctionSpy, shouldMatch: boolean, args?: Array<any>) {
+
+    super(`function was ${shouldMatch && !(args && actualValue.calls.length) ? "not " : ""}called${args && actualValue.calls.length ? " with " + actualValue.calls.map(call => JSON.stringify(call.args)).join(", ") : ""}.`,
```

Comment:
planning to refactor these as part of a larger refactor to do with printing out results, have moved to after beta.0 as need to get that out as priority :)